### ISSUE TITLE
spago build all subfolders

### ISSUE THREAD

**Issue by joprice** [purescript/spago#692]
When using a "monorepo", I expected `spago build` to build all subfolders recursively, but instead I get this error:

```
[error] Failed to read the config. Error was:
[error] There's no "spago.dhall" in your current location.

If you already have a spago project you might be in the wrong subdirectory,
otherwise you might want to run `spago init` to initialize a new project.
```

Am I misconfiguring something, or is this not a supported feature?

**Comment by JordanMartinez** [comment]
I might be wrong, but generally you will use multiple `spago.dhall` files and then run `spago -x config.dhall build`

Referring to [the Monorepo section in the readme](https://github.com/purescript/spago#monorepo), you would ues `spago -x app1/spago.dhall build`

**Comment by f-f** [comment]
We support running spago only in folders where there's a `spago.dhall`, or if you point it to a different config file with the `-x` flag. What does your monorepo look like, and what would you like to achieve?

**Comment by joprice** [comment]
It's just a few separate libraries and binaries. I expected this functionality to exist since other build tools that support multiple modules like maven, sbt, cargo, bazel, etc support building all sub-projects with a single command. It can be useful for ci and quickly validating that dependency changes didn't break things. For instance, after making changes to a module used by multiple libraries in sbt, I would run "compile" and everything that depends on it would be incrementally rebuilt.

I ended up writing a bash script to find the spago.dhall files and build the subprojects in one go. But I wonder if I could just define a dummy project that depends on all of the modules? I'll try that out as well.

**Comment by f-f** [comment]
I see now, thanks for the clarification. A dummy project will work, but this will be fixed once we implement #681, that allows a config file to have multiple "build targets". At that point we could provide a flag to build all the targets of a config, or something like that.

**Comment by ad-si** [comment]
Just stumbled upon this issue as well. I think `spago build` should definitely do something in the monorepo root. Can it maybe at least print a warning like

> You must run `spago --config sub-package/spago.dhall build` to build included packages
